store/tikv/gc_worker.go

```diff
@@ -43,10 +44,37 @@ type GCWorker struct {
 	lastFinish  time.Time
 	cancel      goctx.CancelFunc
 	done        chan error
+
+	session tidb.Session
+
+	spsession   tidb.Session
+	sessionLock sync.Mutex
+}
+
+// StartSafePointChecker triggers SafePoint Checker on each tidb
+func (w *GCWorker) StartSafePointChecker() {
+	w.spsession = createSession(w.store)
+	go func() {
+		for {
+			select {
+			case <-w.store.spMsg:
```

Comment:
Don't use sleep, use 
```
ticker := time.NewTicker()
select {
    case <-w.store.spMsg:
    case <-ticker.C:
}
```
When you sleep, you'll miss signal from `w.store.spMsg`

Reply:
OK, I have revised.